## Step 0: Setup

In [3]:
from sklearn.datasets import load_iris  # dataset
from sklearn.linear_model import LogisticRegression  # supervised classification model
from sklearn.model_selection import KFold, cross_val_score  # for kfold cross validation and the performance scores

## Step 1: load dataset

In [4]:
iris = load_iris()

In [6]:
for key in iris:
    print(key)

data
target
frame
target_names
DESCR
feature_names
filename
data_module


In [5]:
X = iris.data
y = iris.target

In [17]:
# see the data as dataFrame

import pandas as pd
import numpy as np
df = pd.DataFrame(X, columns = iris.feature_names)
df['target'] = y
df.head()

sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)  \
0                5.1               3.5                1.4               0.2   
1                4.9               3.0                1.4               0.2   
2                4.7               3.2                1.3               0.2   
3                4.6               3.1                1.5               0.2   
4                5.0               3.6                1.4               0.2   

   target  
0       0  
1       0  
2       0  
3       0  
4       0

## Step 2: Initialise Classification model 
here Logistic Regression

In [20]:
log_reg_model = LogisticRegression(max_iter = 1000)

## Step 3: K folds Cross Validation

In [19]:
# initialize kfolds object
kf = KFold( n_splits = 5 )

we can also set other parameters here like : 
* `shuffle` : a boolean parameter, True by default, if True it shuffles the datapoint, Note: shuffle = False in TimeSeries
* `random_state` : random seed for shuffling

In [23]:
# use the utility function cross_val_score which automates training/testing across folds and returns performance scores.
scores = cross_val_score(
    estimator = log_reg_model,   # model
    X = X,     # samples
    y = y,     # target
    cv = kf,   # cross validation technique
    scoring = "accuracy"  # performance metric
)

In [32]:
print(type(scores))

<class 'numpy.ndarray'>


## Step 4: Results

In [29]:
print("Cross Validation Scores (Accuracy each fold) : ", scores)
for score in scores:
    print(f"{score:.2f}")

Cross Validation Scores (Accuracy each fold) :  [1.         1.         0.86666667 0.93333333 0.83333333]
1.00
1.00
0.87
0.93
0.83


In [33]:
print("Mean Accuracy : ", scores.mean())

Mean Accuracy :  0.9266666666666665


So Average Accuracy with **Logistic Regression** on this dataset is 92.67 %